File: python/api/pocapi/storage.py
from __future__ import annotations

import requests

from pocapi.settings import POCAPI_URL


def avg_price(
    groupby: str,
    field: str,
    input_path: str,
    output_path: str,
) -> bool:
    url = f"{POCAPI_URL}/storage/avg-price"
    response = requests.post(
        url,
        json={
            "groupby": groupby,
            "field": field,
            "input_path": input_path,
            "output_path": output_path,
        },
    )
    return response.json()["status"]

File: r/api/src/storage.R
POCAPI_URL <- Sys.getenv("POCAPI_URL", unset="http://localhost:6666")

library(httr)

avg_price <- function(groupby, field, input_path, output_path){
    url <- paste(POCAPI_URL, "/storage/avg-price", sep="")
    input_data <- list(
        groupby = groupby,
        field = field,
        input_path=input_path,
        output_path=output_path
    )
    r <- POST(url, body = input_data, encode = "json")
    return(content(r, as="parsed"))
}
